cache/cache.go

```diff
@@ -0,0 +1,97 @@
+package cache
+
+import (
+	"io"
+
+	"github.com/drone-plugins/drone-cache/archive"
+	"github.com/drone-plugins/drone-cache/storage"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type Build struct {
+	Owner  string
+	Repo   string
+	Branch string
+}
+
+type Cache struct {
+	s storage.Storage
+}
+
+func NewCache(s storage.Storage) (Cache, error) {
```

Comment:
You can call this `New()` since cache is implied in the package name, ie `cache.New()`